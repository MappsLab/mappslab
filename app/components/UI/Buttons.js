// @flow
import React from 'react'
import styled from 'styled-components'

import { buttonStyles } from './styles'

export const ButtonWrapper = styled.button`
	${buttonStyles};
`

type ButtonProps = {
	type?: 'button' | 'submit' | 'reset',
}

export const Button = (props: ButtonProps) => <ButtonWrapper {...props} />

Button.defaultProps = {
	type: 'button',
}

type SubmitButtonProps = {
	disabled: boolean,
}

export const SubmitButton = (props: SubmitButtonProps) => <ButtonWrapper {...props} type="submit" />
