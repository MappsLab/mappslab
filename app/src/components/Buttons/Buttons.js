// @flow
import React from 'react'
import styled, { css } from 'styled-components'
import { Link } from 'react-router-dom'

export const ButtonWrapper = styled.button`
	${({ theme, level, disabled }) => css`
		display: inline-block;
		cursor: pointer;
		margin: ${level === 'tertiary' ? theme.layout.spacing.quarter : theme.layout.spacing.single} auto;	
		padding: ${level === 'tertiary' ? theme.layout.spacing.half : theme.layout.spacing.single};
		font-size: ${level === 'tertiary' ? theme.font.size.h5 : theme.font.size.p};
		font-weight: ${level === 'primary' ? theme.font.weight.semi : theme.font.weight.normal};
		border-radius: ${theme.mixins.borderRadius};
		box-shadow: ${level === 'tertiary' ? '' : theme.mixins.boxShadow.normal};
		background-color: ${level === 'primary' ? theme.color.primary.normal : 'white'};
		color: ${
			level === 'primary'
				? theme.color.primary.text
				: level === 'secondary '
				? theme.color.primary.normal
				: theme.color.middleGray
		};
		border: ${level === 'secondary' ? `1px solid ${theme.color.primary.normal}` : ''};
		transition: 0.1s;
		opacity: ${disabled ? 0.6 : 1};
		pointer-events: ${disabled ? 'none' : 'auto'};


		&:hover {
			background-color: ${level === 'primary' ? theme.color.primary.accent : ''};
			box-shadow: ${level === 'tertiary' ? theme.mixins.boxShadow.normal : ''};

		`};
	}
`

type ButtonProps = {
	type?: 'button' | 'submit' | 'reset',
	level?: 'primary' | 'secondary' | 'tertiary',
	onClick?: () => any | (() => Promise<any>),
	to?: string,
	as?: string,
}

export const Button = (props: ButtonProps) => (
	<ButtonWrapper {...props} as={props.as || (props.to ? Link : undefined)} type={props.to || props.as ? undefined : props.type} />
)

Button.defaultProps = {
	type: 'button',
	level: 'primary',
	onClick: undefined,
	to: undefined,
	as: undefined,
}

type SubmitButtonProps = {
	disabled: boolean,
}

export const SubmitButton = (props: SubmitButtonProps) => <ButtonWrapper {...props} type="submit" level="primary" />
