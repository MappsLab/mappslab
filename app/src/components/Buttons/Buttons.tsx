import React from 'react'
import styled, { css, DefaultTheme } from 'styled-components'
import { Link } from 'react-router-dom'
import { ToolTip } from '../ToolTip'

interface ButtonWrapperProps {
	theme: DefaultTheme
	disabled: boolean
	level: string
}

export const ButtonWrapper = styled.button`
	${({ theme, level, disabled }: ButtonWrapperProps) => css`
		display: inline-flex;
		align-items: center;
		cursor: pointer;
		margin: ${level === 'tertiary'
				? theme.layout.spacing.quarter
				: theme.layout.spacing.single}
			auto;
		padding: ${level === 'tertiary'
			? theme.layout.spacing.half
			: theme.layout.spacing.single};
		font-size: ${level === 'tertiary' ? theme.font.size.h5 : theme.font.size.p};
		font-weight: ${level === 'primary'
			? theme.font.weight.semi
			: theme.font.weight.normal};
		border-radius: ${theme.mixins.borderRadius};
		box-shadow: ${level === 'tertiary' ? '' : theme.mixins.boxShadow.normal};
		background-color: ${level === 'primary'
			? theme.color.primary.normal
			: 'white'};
		color: ${level === 'primary'
			? theme.color.primary.text
			: level === 'secondary '
			? theme.color.primary.normal
			: theme.color.middleGray};
		border: ${level === 'secondary'
			? css`1px solid ${theme.color.primary.normal}`
			: ''};
		transition: 0.1s;
		${disabled &&
			css`
				opacity: 0.6;
				pointer-events: none;
			`}

		& > svg {
			margin: 0 0.5em;
		}

		&:hover {
			background-color: ${level === 'primary'
				? theme.color.primary.accent
				: ''};
			box-shadow: ${level === 'tertiary' ? theme.mixins.boxShadow.normal : ''};
		}
	`}
`

interface ButtonProps {
	type?: string
	level?: string
	onClick?: () => any | (() => Promise<any>)
	tooltip?: string
	to?: string
	as?: string
	children?: React.ReactNode
	disabled?: boolean
	htmlFor?: string
}

export const Button = (props: ButtonProps) => (
	<ToolTip message={props.tooltip}>
		<ButtonWrapper
			{...props}
			// @ts-ignore
			as={props.as || (props.to ? Link : undefined)}
			type={props.to || props.as ? undefined : props.type}
		/>
	</ToolTip>
)

Button.defaultProps = {
	type: 'button',
	level: 'primary',
	tooltip: undefined,
	onClick: undefined,
	to: undefined,
	as: undefined,
}

interface SubmitButtonProps extends ButtonProps {
	wide: boolean
}

export const SubmitButton = ({ as, ...rest }: SubmitButtonProps) => (
	<ButtonWrapper
		// @ts-ignore
		as={as}
		{...rest}
		type="submit"
		level="primary"
	/>
)
