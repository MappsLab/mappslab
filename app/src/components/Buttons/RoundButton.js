// @flow
import * as React from 'react'
import styled from 'styled-components'
import ToolTip from 'Components/ToolTip'

const Wrapper = styled.button`
	border: 2px solid ${({ active }) => (active ? 'red' : 'green')};
	display: flex;
	justify-content: center;
	align-items: center;
	text-align: center;
	${({ size }) =>
		size === 'large'
			? `
		width: 50px;
		height: 50px;
		border-radius: 25px;
		margin: 5px;
		font-size: 35px;
	`
			: `
		width: 35px;
		height: 35px;
		border-radius: 18px;
		margin: 4px;
		font-size: 22px;
	`};
`

/**
 * RoundButton
 */

type BaseProps = {
	onClick: () => void,
	isActive: boolean,
	disabled: boolean,
	label: string,
	icon: React.Node,
	size: 'normal' | 'large',
}

const RoundButtonBase = ({ onClick, isActive, label, disabled, icon, size }: BaseProps) => (
	<Wrapper size={size} onClick={onClick} active={isActive} disabled={disabled}>
		<span role="img" aria-label={label}>
			{icon}
		</span>
	</Wrapper>
)

type Props = {
	onClick: () => void,
	isActive?: boolean,
	disabled?: boolean,
	label: string,
	icon: string,
	size?: 'normal' | 'large',
	toolTip?: string,
}

const RoundButton = ({ toolTip, ...rest }: Props) =>
	toolTip && toolTip.length ? (
		<ToolTip message={toolTip}>
			<RoundButtonBase {...rest} />
		</ToolTip>
	) : (
		<RoundButtonBase {...rest} />
	)

RoundButton.defaultProps = {
	isActive: false,
	disabled: false,
	size: 'normal',
	toolTip: undefined,
}

export default RoundButton
