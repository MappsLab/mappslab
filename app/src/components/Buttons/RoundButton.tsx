import * as React from 'react'
import styled, { css } from 'styled-components'
import { ToolTip } from 'Components/ToolTip'

interface WrapperProps {
	active?: boolean
	size?: string
}

const Wrapper = styled.button`
	${({ active, size }: WrapperProps) => css`
		border: 2px solid ${active ? 'red' : 'green'};
		display: flex;
		justify-content: center;
		align-items: center;
		text-align: center;
		${size === 'large'
			? css`
					width: 50px;
					height: 50px;
					border-radius: 25px;
					margin: 5px;
					font-size: 35px;
			  `
			: css`
					width: 35px;
					height: 35px;
					border-radius: 18px;
					margin: 4px;
					font-size: 22px;
			  `}
	`}
`

/**
 * RoundButton
 */

interface BaseProps {
	onClick: () => void
	isActive?: boolean
	disabled?: boolean
	label: string
	icon: React.ReactNode
	size?: 'normal' | 'large'
	tooltip?: string
}

export const RoundButton = ({
	onClick,
	isActive,
	label,
	disabled,
	icon,
	size,
	tooltip,
}: BaseProps) => (
	<ToolTip message={tooltip}>
		<Wrapper
			size={size}
			onClick={onClick}
			active={isActive}
			disabled={disabled}
		>
			<span role="img" aria-label={label}>
				{icon}
			</span>
		</Wrapper>
	</ToolTip>
)

RoundButton.defaultProps = {
	isActive: false,
	disabled: false,
	size: 'normal',
	toolTip: undefined,
}
