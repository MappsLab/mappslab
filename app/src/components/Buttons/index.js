export * from './Buttons'
export { default as RoundButton } from './RoundButton'
