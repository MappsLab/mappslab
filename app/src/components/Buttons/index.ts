export * from './Buttons'
export * from './RoundButton'
