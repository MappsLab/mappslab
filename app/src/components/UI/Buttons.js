// @flow
import React from 'react'
import styled from 'styled-components'

export const ButtonWrapper = styled.button`
	${({ theme, level }) => `
		display: block;
		margin: ${level === 'tertiary' ? theme.layout.spacing.quarter : theme.layout.spacing.single} auto;	
		padding: ${level === 'tertiary' ? theme.layout.spacing.half : theme.layout.spacing.single};
		font-size: ${level === 'tertiary' ? theme.text.size.h5 : theme.text.size.p};
		font-weight: ${level === 'primary' ? theme.text.weight.semi : theme.text.weight.normal};
		border-radius: ${theme.mixins.borderRadius};
		box-shadow: ${level === 'tertiary' ? '' : theme.mixins.boxShadow.normal};
		background-color: ${level === 'primary' ? theme.color.primary.normal : 'white'};
		color: ${
			level === 'primary'
				? theme.color.primary.text
				: level === 'secondary '
					? theme.color.primary.normal
					: theme.color.middleGray
		};
		border: ${level === 'secondary' ? `1px solid ${theme.color.primary.normal}` : ''};
		transition: 0.1s;

		&:hover {
			background-color: ${level === 'primary' ? theme.color.primary.accent : ''};
			box-shadow: ${level === 'tertiary' ? theme.mixins.boxShadow.normal : ''};

		`};
	}
`

type ButtonProps = {
	type?: 'button' | 'submit' | 'reset',
	level?: 'primary' | 'secondary' | 'tertiary',
}

export const Button = (props: ButtonProps) => <ButtonWrapper {...props} />

Button.defaultProps = {
	type: 'button',
	level: 'primary',
}

type SubmitButtonProps = {
	disabled: boolean,
}

export const SubmitButton = (props: SubmitButtonProps) => <ButtonWrapper {...props} type="submit" level="primary" />
