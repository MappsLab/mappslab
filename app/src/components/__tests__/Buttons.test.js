/* eslint-disable no-undef */
import React from 'react'
import { render } from 'react-testing-library'

import { Button } from 'Components/Buttons'

const buttonText = 'Click Me!'

describe('Button Component', () => {
	it('Mounts Correctly', () => {
		const { container } = render(<Button>{buttonText}</Button>)
		expect(container).toMatchSnapshot()
	})
})
