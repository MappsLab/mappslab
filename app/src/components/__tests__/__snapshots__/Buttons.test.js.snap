// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Mounts Correctly 1`] = `
<Buttons__ButtonWrapper
  type="button"
>
  Click Me!
</Buttons__ButtonWrapper>
`;
